docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,90 @@
+
+Let’s look at a specific example with testing expired metadata. We are going to look in more detail at various edge cases bellow, but now let’s just say that in this test we use two of the cool things the RepositorySimulator provides:
+* Upgrade [targets](https://theupdateframework.github.io/specification/latest/#targets) to v2
+* Change [timestamp](https://theupdateframework.github.io/specification/latest/#timestamp) v2 expiry date
+
+So how can we verify that the metadata looks as expected, without the need to implement file access?
+
+First, we can use the `--dump` option to find the corresponding temporary directory:
+
+```
+$ python3 test_updater_top_level_update.py TestRefresh.test_expired_metadata --dump
+Repository Simulator dumps in /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_
+```
+
+Once we know it, we can verify that the metadata has 2 cached versions:
+```
+$ ls /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/
+1	2
+```
+```
+$ ls /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/1
+1.root.json	snapshot.json	targets.json	timestamp.json
+```
+
+And now we can also see that after bumping the version and moving timestamp v2 expiry date two weeks forward from v1, the v2 corresponding timestamp metadata has recorded that expiry date correctly:
+
+Timestamp v1:
+<pre><code>$ cat /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/1/timestamp.json 
+{
+ "signatures": [{...}],
+ "signed": {
+  "_type": "timestamp",
+  <b>"expires": "2022-03-30T00:18:31Z"</b>,
+  "meta": { "snapshot.json": {"version": 1}},
+  "spec_version": "1.0.28",
+  "version": 1
+ }}
+</code></pre>
+
+Timestamp v2:
+
+<pre><code>$ cat /var/folders/pr/b0xyysh907s7mvs3wxv7vvb80000gp/T/tmpzvr5xah_/test_expired_metadata/2/timestamp.json 
+{
+ "signatures": [{...}],
+ "signed": {
+  "_type": "timestamp",
+  <b>"expires": "2022-04-13T00:18:31Z"</b>,
+  "meta": { "snapshot.json": {"version": 2}},
+  "spec_version": "1.0.28",
+  "version": 2
+ }}
+</code></pre>
+
+## Edge cases are a norm when we talk about security
+
+The TUF spec says the following:
+“Record the time at which the update began as the fixed update start time. Time is fixed at the beginning of the update workflow to allow an application using TUF to effectively pause time, in order to ensure that metadata which has a valid expiration time at the beginning of an update does not fail an expiration check later in the update workflow.”
+
+Now imagine that we have performed an update and stored metadata in cache. And the locally stored timestamp/snapshot has expired. But we still need it to perform an update from remote by verifying the signatures and we need to use the expired timestamp. Here the repository simulator can do a great job, as we’d like to somehow update expiry without explicitly touching the file system from the tests.
```

Comment:
```suggestion
Now imagine that we have performed an update and stored metadata in a cache. And the locally stored timestamp/snapshot has expired. But we still need it to perform an update from remote by verifying the signatures and we need to use the expired timestamp. Here the repository simulator can do a great job testing this scenario, as we’d like to somehow update expiry without explicitly touching the file system from the tests.
```
I think it should be underlined clearer that the `RepositorySimulator` can be useful for testing this scenerio, not actually performing it.